systemtest/src/test/java/io/strimzi/systemtest/upgrade/kraft/KRaftKafkaUpgradeDowngradeST.java

```diff
@@ -0,0 +1,328 @@
+ * Metadata for upgrade/downgrade procedure are loaded from kafka-versions.yaml in root dir of this repository.
+ */
+@Tag(KRAFT_UPGRADE)
+public class KRaftKafkaUpgradeDowngradeST extends AbstractKRaftUpgradeST {
+    private static final Logger LOGGER = LogManager.getLogger(KRaftKafkaUpgradeDowngradeST.class);
+
+    private final String continuousTopicName = "continuous-topic";
+    private final int continuousClientsMessageCount = 1000;
+
+    @IsolatedTest
+    void testKafkaClusterUpgrade(ExtensionContext testContext) {
+        List<TestKafkaVersion> sortedVersions = TestKafkaVersion.getSupportedKafkaVersions();
+
+        String producerName = clusterName + "-producer";
+        String consumerName = clusterName + "-consumer";
+
+        for (int x = 0; x < sortedVersions.size() - 1; x++) {
+            TestKafkaVersion initialVersion = sortedVersions.get(x);
+            TestKafkaVersion newVersion = sortedVersions.get(x + 1);
+
+            // If it is an upgrade test we keep the message format as the lower version number
+            String metadataVersion = initialVersion.metadataVersion();
+
+            runVersionChange(initialVersion, newVersion, producerName, consumerName, metadataVersion, 3, 3, testContext);
+        }
+
+        // ##############################
+        // Validate that continuous clients finished successfully
+        // ##############################
+        ClientUtils.waitForClientsSuccess(producerName, consumerName, TestConstants.CO_NAMESPACE, continuousClientsMessageCount);
+        // ##############################
+    }
+
+    @IsolatedTest
+    void testKafkaClusterDowngrade(ExtensionContext testContext) {
+        final TestStorage testStorage = storageMap.get(testContext);
+        List<TestKafkaVersion> sortedVersions = TestKafkaVersion.getSupportedKafkaVersions();
+
+        String clusterName = testStorage.getClusterName();
+        String producerName = clusterName + "-producer";
+        String consumerName = clusterName + "-consumer";
+
+        for (int x = sortedVersions.size() - 1; x > 0; x--) {
+            TestKafkaVersion initialVersion = sortedVersions.get(x);
+            TestKafkaVersion newVersion = sortedVersions.get(x - 1);
+
+            // If it is a downgrade then we make sure to use the lower version number for the message format
+            String metadataVersion = newVersion.metadataVersion();
+            runVersionChange(initialVersion, newVersion, producerName, consumerName, metadataVersion, 3, 3, testContext);
+        }
+
+        // ##############################
+        // Validate that continuous clients finished successfully
+        // ##############################
+        ClientUtils.waitForClientsSuccess(producerName, consumerName, TestConstants.CO_NAMESPACE, continuousClientsMessageCount);
+        // ##############################
+    }
+
+    @IsolatedTest
+    @Disabled("This is currently not possible - missing support for unsafe downgrades")
```

Comment:
Not sure we really want to have this test -> the operator will not do the unsafe downgrades. The user will have to do them. The _unsafe_ downgrade also means it might not work in some cases.

Reply:
Ah okay, I thought that there will be some kind of option in the future. So I just misread it. Thanks :)